### PR TITLE
formhandler: Add statistics

### PR DESCRIPTION
**Pull request by kingthorin**
- CHANGELOG > Add note.
- ExtensionFormHandler > Add stats constants.
- OptionsFormHandlerPanel > Add stats handling for removals.
- PopupDialogAddField > Add stats handling for additions.
- FormHandlerHelp.html > Added stats info to the help.

image

### CODE REVIEW

**Comment by psiinon** [comment]
And the stats page https://www.zaproxy.org/docs/internal-statistics/ (in another PR;)

File: addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/DialogAddField.java [hunk 1]
```diff
@@ -198,12 +199,15 @@ protected void showBadRegexDialog() {
      */
     @Override
     protected void performAction() {
```
Comment (thc202) [inline addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/DialogAddField.java:201]: This is also called when modifying.

File: addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/OptionsFormHandlerPanel.java [hunk 2]
```diff
@@ -185,7 +190,8 @@ public boolean showRemoveDialogue(FormHandlerParamField e) {
 
             if (option == JOptionPane.OK_OPTION) {
                 setRemoveWithoutConfirmation(removeWithoutConfirmationCheckBox.isSelected());
-
+                Stats.incCounter(ExtensionFormHandler.STATS_REMOVE);
+                Stats.incCounter(ExtensionFormHandler.STATS_REMOVE + "." + field.getName());
```
Comment (thc202) [inline addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/OptionsFormHandlerPanel.java:194]: Could be extracted a method, e.g. `incStat(ExtensionFormHandler.STATS_REMOVE, field)`.

File: addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/DialogAddField.java [hunk 3]
```diff
@@ -198,9 +198,10 @@ protected void showBadRegexDialog() {
...
+        String name = getNameTextField().getText();
         field =
                 new FormHandlerParamField(
-                        getNameTextField().getText(),
+                        name,
```
Comment (thc202) [inline addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/DialogAddField.java:204]: These changes can be reverted.

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**